### PR TITLE
Added Jenkins file and Docker file

### PR DESCRIPTION
**Pull request by saksham000**
1. Updated Application to Spring 3.3.3
2. Added Jenkins and Docker file.
3. Added a Test Case for Testing Jenkins.
4. Updated Readme file.

### CODE REVIEW

File: src/main/java/com/swapnixi/feedbackhub/service/implementation/UserServiceImpl.java [hunk 1]
```diff
@@ -72,16 +72,6 @@ public List<User> getUsersWithNameContainingNewKeyword() {
         return  userRepository.nameHasNewKeyword();
     }
```
Comment (saksham000) [inline src/main/java/com/swapnixi/feedbackhub/service/implementation/UserServiceImpl.java:96]: This Function was not in use and giving a warning.

Comment (swapnilxi) [reply src/main/java/com/swapnixi/feedbackhub/service/implementation/UserServiceImpl.java:96]: Please review its implementation; if it serves no purpose, we can remove it. However, I believe it may be useful. Also, examine the user-ent branch, which requires improvements. If you can improve it, then proceed with the merge.

File: src/main/java/com/swapnixi/feedbackhub/service/implementation/UserServiceImpl.java [hunk 2]
```diff
@@ -72,16 +72,6 @@ public List<User> getUsersWithNameContainingNewKeyword() {
...
-    private User DtoToUser(UserDTO userDTO) {
-        User user = new User();
-        user.setId(userDTO.getId());
-        user.setName(userDTO.getName());
-        user.setEmail(userDTO.getEmail());
-        user.setAbout(userDTO.getAbout());
-        user.setPassword(userDTO.getPassword());
-        return user;
-    }
-
```
Comment (swapnilxi) [inline src/main/java/com/swapnixi/feedbackhub/service/implementation/UserServiceImpl.java:84]: do not remove this, it was for the reason to handle database queries , so you can revert this change

File: src/main/java/com/swapnixi/feedbackhub/controller/UserController.java [hunk 3]
```diff
@@ -19,8 +21,22 @@ public class UserController {
...
     public ResponseEntity<User> createUser(@RequestBody User user) {
-        User createdUser = userService.createUser(user);
-        return ResponseEntity.status(HttpStatus.CREATED).body(createdUser);
+        try {
+            User createdUser = userService.createUser(user);
+            return ResponseEntity.status(HttpStatus.CREATED).body(createdUser);
+        } catch (UserAlreadyRegestredException e) {
+            throw new ResponseStatusException(HttpStatus.NOT_ACCEPTABLE, e.getMessage());
+        }
+    }
```
Comment (saksham000) [inline src/main/java/com/swapnixi/feedbackhub/controller/UserController.java:30]: This will Check whether the user have already registered or not. This will solve redundancy problem.

File: src/main/java/com/swapnixi/feedbackhub/controller/UserController.java [hunk 4]
```diff
@@ -19,8 +21,22 @@ public class UserController {
...
+
+    @GetMapping(path = "login/{email}/{password}")
+    public boolean userLogin(@PathVariable String email, @PathVariable String password) {
+        if (userService.loginUser(email, password)) {
+            ResponseEntity.accepted();
+            return true;
+        } else {
+            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, "Invalid Credentials");
+        }
```
Comment (saksham000) [inline src/main/java/com/swapnixi/feedbackhub/controller/UserController.java:39]: Users can login with valid credential's only, This API will take email and password as parameter.

File: src/main/java/com/swapnixi/feedbackhub/service/implementation/UserServiceImpl.java [hunk 5]
```diff
@@ -19,10 +23,24 @@ public class UserServiceImpl implements UserService {
 
     @Override
     public User createUser(User user) {
-//        User user = DtoToUser(user);
-        User savedUser = userRepository.save(user);
-//        return userToDTO(savedUser);
-        return  savedUser;
```
Comment (saksham000) [inline src/main/java/com/swapnixi/feedbackhub/service/implementation/UserServiceImpl.java:25]: Before this API registers the new user without checking whether the user is already registered or not which causes redundancy problem during login.

File: src/main/java/com/swapnixi/feedbackhub/service/implementation/UserServiceImpl.java [hunk 6]
```diff
@@ -64,22 +82,16 @@ public List<User> getAllUsers() {
...
-    private User DtoToUser(UserDTO userDTO) {
-        User user = new User();
-        user.setId(userDTO.getId());
-        user.setName(userDTO.getName());
-        user.setEmail(userDTO.getEmail());
-        user.setAbout(userDTO.getAbout());
-        user.setPassword(userDTO.getPassword());
-        return user;
+        return userRepository.nameHasNewKeyword();
     }
```
Comment (saksham000) [inline src/main/java/com/swapnixi/feedbackhub/service/implementation/UserServiceImpl.java:95]: I have not removed this function, I dont know why this is showing that I have removed this function.

File: README.md [hunk 7]
```diff
@@ -171,4 +171,60 @@ spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.PostgreSQLDialect
...
+            steps {
+                script {
+                    bat 'docker build -t {DockerHub Repo name}:feedbackhub-v1 .'
+                }
+            }
+        }
+        stage('Push Docker image to Hub') {
+            steps {
+                script {
+                    withCredentials([string(credentialsId: '{Variabel Id}', variable: '{DockerHub Password Stored in Secrete Key use Jenkins Syntax Creator for Key}')]) {
```
Comment (saksham000) [inline README.md:205]: Here I have explained that use Jenkins to create variable to store Password.

File: src/main/java/com/swapnixi/feedbackhub/service/implementation/UserServiceImpl.java [hunk 8]
```diff
@@ -64,22 +82,16 @@ public List<User> getAllUsers() {
     public void deleteUser(Long userId) {
         // Implementation for deleting a user by ID
         // This method should delete the user with the given ID from the repository
-        throw new UnsupportedOperationException("Unimplemented method 'deleteUser'");
```
Comment (saksham000) [inline src/main/java/com/swapnixi/feedbackhub/service/implementation/UserServiceImpl.java:67]: You have not implemented the Delete logic. So I have Implemented it.
